src/irr/asset/CGraphicsPipelineLoaderMTL.cpp

```diff
@@ -257,47 +289,18 @@ layout (set = 3, binding = 6) uniform sampler2D map_bump;
 #endif //_IRR_FRAG_SET3_BINDINGS_DEFINED_
 
 #include <irr/builtin/glsl/bsdf/brdf/specular/fresnel/fresnel.glsl>
-
-#ifndef _IRR_COMPUTE_LIGHTING_DEFINED_
-#define _IRR_COMPUTE_LIGHTING_DEFINED_
-
-irr_glsl_BSDFIsotropicParams irr_glsl_computeLighting()
-{
-    //would use irr_glsl_calcFragmentShaderSurfaceInteraction(), but then dFd_(ViewPos) would be computed twice
-    irr_glsl_ViewSurfaceInteraction interaction;
-    interaction.V.dir = -ViewPos;
-    interaction.V.dPosdScreen[0] = dFdx(ViewPos);
-    interaction.V.dPosdScreen[1] = dFdy(ViewPos);
-    interaction.N = normalize(Normal);
-
-    if ((PC.params.extra&map_bump_MASK) == map_bump_MASK)
-    {
-        float height = texture(map_bump, UV).x;
-        vec3 dpdx = interaction.V.dPosdScreen[0];
-        vec3 dpdy = interaction.V.dPosdScreen[1];
-        float dhdx = dFdx(height);
-        float dhdy = dFdy(height);
-        
-        vec3 r1 = cross(dpdy, interaction.N);
-        vec3 r2 = cross(interaction.N, dpdx);
-        vec3 surfGrad = (r1*dhdx + r2*dhdy) / dot(dpdx,r1);
-        interaction.N = normalize(interaction.N - surfGrad);
-    }
-    return irr_glsl_calcBSDFIsotropicParams(interaction, -ViewPos);
-}
-#endif //_IRR_COMPUTE_LIGHTING_DEFINED_
-
+#include <irr/builtin/glsl/bsdf/brdf/diffuse/fresnel_correction.glsl>
+#include <irr/builtin/glsl/bsdf/brdf/diffuse/lambert.glsl>
 
 #ifndef _IRR_BSDF_COS_EVAL_DEFINED_
 #define _IRR_BSDF_COS_EVAL_DEFINED_
 
 // Spectrum can be exchanged to a float for monochrome
 #define Spectrum vec3
-#define Ia 0.1
 
 //! This is the function that evaluates the BSDF for specific view and observer direction
 // params can be either BSDFIsotropicParams or BSDFAnisotropicParams 
-Spectrum irr_glsl_bsdf_cos_eval(in irr_glsl_BSDFIsotropicParams params)
+Spectrum irr_bsdf_cos_eval(in irr_glsl_BSDFIsotropicParams params)
```

Comment:
maybe you wouldn't need two reimplementations of `irr_bsdf_cos_eval` if you created an `irr_fetch_material_parameters` and delegated the choice of fetching `Ka,Ks,Ns` from texture or constants and normal/bump mapping to that function instead

Reply:
solved with a simple `#ifndef _NO_UV`